src/capture.h

```diff
@@ -0,0 +1,35 @@
+#ifndef CAPTURE_H
+#define CAPTURE_H
+
+#include <bcm_host.h>
+
+#define MAX_REQUEST_BUFFER_SIZE     256
+
+struct capture_info {
+    int display_id;
+    int display_width;
+    int display_height;
+
+    int capture_width;
+    int capture_height;
+    int capture_stride;
+
+    uint16_t mono_threshold_r5;
+    uint16_t mono_threshold_g6;
+    uint16_t mono_threshold_b5;
+
+    uint16_t *buffer;
+    uint8_t *work;
+
+    DISPMANX_DISPLAY_HANDLE_T display_handle;
+    DISPMANX_RESOURCE_HANDLE_T capture_resource;
+
+    uint8_t request_buffer[MAX_REQUEST_BUFFER_SIZE];
+    int request_buffer_ix;
+
+    int send_snapshot;
+
+    int dithering;
+    int16_t *dithering_buffer;
```

Comment:
I'm looking through the code and it seems that `dithering_buffer` is a `uint8_t *`. Is this true? I didn't see anything negative?

Reply:
The error values sometimes can be negative, and are added to the values in the buffer. I'm not sure if could work with `uint8_t`.

Reply:
Ok, the intermediate type will need to be `int16` (or just `int`) for that to work, but it sounds like what gets saved back is `uint8_t`. 

We should be able to verify that the `uint8_t` switch works via the unit tests. Btw, in case you're wondering why I'm slowing things down to get the unit tests in, the lack of unit tests had been bothering me for a while. The addition of dithering raise the amount of time I need to devote to this library for maintenance enough that I think it's worth getting in now.